awx_collection/tools/roles/generate/templates/tower_module.j2

```diff
@@ -175,9 +175,8 @@ def main():
 {% endfor %}
 
     # Attempt to look up an existing item based on the provided data
-    existing_item = module.get_one('{{ item_type }}', **{
+    existing_item, name = module.get_one('{{ item_type }}', name_or_id={{ name_option }}, **{
```

Comment:
Again, I think you dropped the 2 returned items so this needs to be updated

Reply:
Updated and also fixed the call for name in the data structure down below for the new `new_name if new_name else (module... if existing_object else {{ name_field }}`